runtime/buffer.h

```diff
@@ -13,7 +13,11 @@
 
 namespace slinky {
 
-using index_t = std::ptrdiff_t;
+#ifdef __APPLE__
+  using index_t = std::int64_t;
```

Comment:
Maybe we just need to `#include <cstddef>`? If we were compiling 32-bit on OSX I think this would be wrong.

Reply:
The problem I am trying to solve here is:

```
error: conversion from 'index_t' (aka 'long') to 'expr' is ambiguous
./runtime/expr.h:189:3: note: candidate constructor
  expr(std::int64_t x);
  ^
./runtime/expr.h:190:3: note: candidate constructor
  expr(std::int32_t x) : expr(static_cast<std::int64_t>(x)) {}
```

Reply:
I think we need to try to solve the ambiguity. I think maybe `expr(const base_expr_node*)` should be a static helper function instead of a constructor, and maybe we should also get rid of the int32_t overload?

Reply:
> If we were compiling 32-bit on OSX

FWIW, neither macOS nor iOS support 32-bit anymore (and haven't for a while), but I agree that adding the #include is a much better fix

Reply:
In that case, maybe this PR is fine, we should just add a static_assert that `sizeof(index_t) == sizeof(std::size_t)` I tried reducing the ambiguity here, it's not easy.

Reply:
> > If we were compiling 32-bit on OSX
> 
> FWIW, neither macOS nor iOS support 32-bit anymore (and haven't for a while), but I agree that adding the #include is a much better fix

Unfortunately, #include <cstddef> doesn't fix the problem